src/iface/ethernet.rs

```diff
@@ -667,14 +684,93 @@ impl<'b, 'c> InterfaceInner<'b, 'c> {
         }
     }
 
+    /// Process an IPv4 fragment
+    /// Returns etiher an assembled packet, or nothing (if no packet is available),
+    /// or an error caused by processing a packet.
+    #[cfg(feature = "fragmentation-ipv4")]
+    fn process_ipv4_fragment<'frame, 'r>
+				    (&mut self,
+				    	ipv4_packet: Ipv4Packet<&'frame [u8]>,
+                        timestamp: Instant,
+				    	fragments: &'r mut Option<FragmentsSet<'e>>
+				    ) ->
+                     Result<Option<Ipv4Packet<&'r [u8]>>> 
+    {
+        match *fragments {
+        	Some(ref mut fragments) => {
+                // get an existing fragment or attempt to get a new one
+				let fragment = match fragments.get_packet(ipv4_packet.ident(), timestamp) {
+				    Some(frag) => frag,
+				    None => return Ok(None), // TODO: throw an exception?
+				};
+
+				if fragment.is_empty() {
+				    // this is a new packet
+				    fragment.set_id(ipv4_packet.ident());
+				}
+
+				if !ipv4_packet.more_frags() {
+				    // last fragment, remember data length
+				    fragment.set_total_len(ipv4_packet.frag_offset() as usize +
+                                           ipv4_packet.total_len() as usize);
+				}
+
+                fragment.add(ipv4_packet.header_len() as usize,
+                             ipv4_packet.frag_offset() as usize,
+                             ipv4_packet.payload().len(),
+                             ipv4_packet.into_inner(),
+                             timestamp)
+                .unwrap();
+
+                if fragment.check_contig_range() {
+                    // this is the last packet, attempt reassembly
+                    let front = fragment.front().unwrap();
+                    {
+                        // because the different mutability of the underlying buffers, we have to do this exercise
+                        let mut ipv4_packet = Ipv4Packet::new_checked(&mut fragment.rx_buffer[0..front])?;
+                        ipv4_packet.set_total_len(front as u16);
+                        ipv4_packet.fill_checksum();
+                    }
+                    fragment.reset();
+                    return Ok(Some(Ipv4Packet::new_checked(&fragment.rx_buffer[0..front])?));
+                }
+
+                // not the last fragment
+                return Ok(None);
+        	},
+        	None => {
+        		return Ok(None); // TODO: throw an exception?
```

Comment:
Also log it, with a different message indicating that fragmentation is not supported.

Reply:
Fixed